tests/unit/handlers/guildRoleHandler.test.ts

```diff
@@ -0,0 +1,253 @@
+    getGuildRolesSpy.mockResolvedValue([]);
+    const mockRequest = generateDummyRequestObject({
+      url: "/roles",
+      headers: { Authorization: "Bearer testtoken" },
+    });
+
+    const response: JSONResponse = await getGuildRolesHandler(
+      mockRequest,
+      guildEnv
+    );
+    const jsonResponse: { roles: Array<GuildRole> } = await response.json();
+    expect(response.status).toBe(200);
+    expect(Array.isArray(jsonResponse.roles)).toBeTruthy();
+    expect(jsonResponse.roles.length).toBe(0);
+  });
+
+  it("should return array of id and name of roles present in guild", async () => {
+    const expectedResponse = rolesMock;
+    getGuildRolesSpy.mockResolvedValueOnce(expectedResponse);
+
+    const mockRequest = generateDummyRequestObject({
+      url: "/roles",
+      headers: { Authorization: "Bearer testtoken" },
+    });
+
+    const response: JSONResponse = await getGuildRolesHandler(
+      mockRequest,
+      guildEnv
+    );
+    const jsonResponse: { roles: Array<GuildRole> } = await response.json();
+    expect(response.status).toBe(200);
+    expect(Array.isArray(jsonResponse.roles)).toBeTruthy();
+    expect(jsonResponse.roles).toEqual(expectedResponse);
+  });
+});
+
+describe("get role by role name", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it("should return a instance of JSONResponse", async () => {
+    getGuildRoleByNameSpy.mockResolvedValueOnce(undefined);
+    const mockRequest = generateDummyRequestObject({
+      url: "/roles",
+      params: {
+        roleName: "everyone",
+      },
+      headers: { Authorization: "Bearer testtoken" },
+    });
+    const response = await getGuildRoleByRoleNameHandler(mockRequest, guildEnv);
+    expect(response).toBeInstanceOf(JSONResponse);
+  });
+
+  it("should return Bad Signature object if no auth headers provided", async () => {
+    getGuildRoleByNameSpy.mockResolvedValueOnce(undefined);
+    const mockRequest = generateDummyRequestObject({
+      url: "/roles",
+      params: {
+        roleName: "everyone",
+      },
+    });
+    const response: JSONResponse = await getGuildRoleByRoleNameHandler(
+      mockRequest,
+      guildEnv
+    );
+    const jsonResponse: { error: string } = await response.json();
+    expect(response.status).toBe(401);
+    expect(jsonResponse).toEqual(responseConstants.BAD_SIGNATURE);
+  });
+
+  it("should return Not Found error if no roleName is not provided", async () => {
+    getGuildRoleByNameSpy.mockResolvedValueOnce(undefined);
+    const mockRequest = generateDummyRequestObject({
+      url: "/roles",
+      params: {},
+
+      headers: { Authorization: "Bearer testtoken" },
+    });
+    const response: JSONResponse = await getGuildRoleByRoleNameHandler(
+      mockRequest,
+      guildEnv
+    );
+    const jsonResponse: { error: string } = await response.json();
+    expect(response.status).toBe(404);
+    expect(jsonResponse).toEqual(responseConstants.NOT_FOUND);
+  });
```

Comment:
I think this is a bad request it should return 400 
not passing a required parameter comes under bad request.

Reply:
Fixed.